lib/Containers/MerkleTree.cpp

```diff
@@ -723,8 +719,7 @@ template<typename Hasher, std::uint64_t const BranchingBits>
 std::vector<std::pair<std::uint64_t, std::uint64_t>>
 MerkleTree<Hasher, BranchingBits>::diff(
     MerkleTree<Hasher, BranchingBits>& other) {
-  std::shared_lock<std::shared_mutex> guard1(_dataLock);
-  std::shared_lock<std::shared_mutex> guard2(other._dataLock);
+  std::scoped_lock guard(_dataLock, other._dataLock);
```

Comment:
Maybe some order? Maybe based on address?
Because std::lock algorithm in libstdc++ based on loop with try_locks

Reply:
The whole point of using `scoped_lock` is to acquire multiple locks with some algorithm that avoids deadlocks. There is usually a good reason why the libstdc++ implementation looks/works the way it does. For the scoped_lock locking algorithm Howard Hinnant shared some details here: http://howardhinnant.github.io/dining_philosophers.html